central/declarativeconfig/utils/integration_health.go

```diff
@@ -12,13 +12,16 @@ import (
 	"github.com/stackrox/rox/pkg/utils"
 )
 
+// DeclarativeConfigHealthIDPrefix is prefix of all integration health's IDs of declarative configurations.
+const DeclarativeConfigHealthIDPrefix = "declarativeconfig."
```

Comment:
what's the current format of ID? I think we discussed some time ago to switch to UUIDs everywhere and use this type on database. With string prefix it will not be possible? Why not add a field that will explicitly indicate origin?

Reply:
Currently it's not constrained into any type of format: https://github.com/stackrox/stackrox/blob/master/proto/storage/integration_health.proto#L25

I do remember this as well. I think the issue here is that currently the ID of the proto will be taken as PK. In case of notifiers, we will potentially override the notifier (with the introduction of the new declarative config type, probably the ID shouldn't be the PK but rather the constraint of ID + integration health type). This would probably be the best solution, this way it would also be possible in the future to switch to `type: uuid` in the proto without huge changes required within the migration.

I'm not sure how easy it currently is to create constraints on the store, I had to do one for groups and was required to go through some hoops: #5099 (and the follow-up PR there). Not sure some of the previous issues were resolved or not.

I don't think it's an option to add the origin to the integration health proto itself, the ID issue would still exist then. So, either we take the prefix, knowing that in the future once the switch to `type:uuid` may come, we have to face the problem again and most likely go for the unique constraint on ID + integration health type, or we do this from the start.

Seeing the options, I'd prefer to go with the unique constraint since this one will be future proof and we won't have to do the workaround we currently do within the manager's implementation (i.e. prefixing the UUID and handling the edge cases there). I think we also have enough time to get this in 4.1. If we won't make it, then the prefix approach should be straightforward.

WDYT @ivan-degtiarenko ?